_minutes/2023-08-03-wecg.md

```diff
@@ -0,0 +1,123 @@
+
+
+## Agenda: [discussion in #432](https://github.com/w3c/webextensions/issues/432), [github issues](https://github.com/w3c/webextensions/issues)
+
+The meeting will start at 3 minutes after the hour.
+
+ * **Carry-over from previous meetings**
+   * None
+ * **Other new issues**
+   * [Issue 431](https://github.com/w3c/webextensions/issues/431): Proposal: Default to Active Tab in scripting.executeScript()
+   * [Issue 433](https://github.com/w3c/webextensions/issues/433): Inconsistency: Alarms API time slipping
+ * **Open discussion queue (add yourself at the bottom)**
+   * None
+ * **Check-in on ongoing issues**
+   * [Issue 362](https://github.com/w3c/webextensions/issues/362): Proposal: Declarative Cosmetic Rules
+   * [Issue 403](https://github.com/w3c/webextensions/issues/403): Proposal: Ability to insert CSS from content script context
+   * [Issue 394](https://github.com/w3c/webextensions/issues/394): [DNR] Add support for wildcards in initiatorDomains and excludedInitiatorDomains fields
+
+
+## Attendees (sign yourself in)
+
+ 1. Jason Waterman (Mozilla)
+ 2. Simeon Vincent (Unaffiliated)
+ 3. Oliver Dunk (Google)
+ 4. Rob Wu (Mozilla)
+ 5. Dmitriy Seregin (AdGuard)
+ 6. Timothy Hatcher (Apple)
+ 7. Benjamin Bruneau (1Password)
+ 8. Tim Heflin (Keeper)
+ 9. Carlos Jeurissen (Jeurissen Apps)
+ 10. Mukul Purohit (Microsoft)
+ 11. Andrey Meshkov (AdGuard)
+ 12. Maxim Topciu (AdGuard)
+ 13. Matt Gibson (Bitwarden)
+
+
+## Meeting notes
+
+[Issue 431](https://github.com/w3c/webextensions/issues/431): Proposal: Default to Active Tab in scripting.executeScript()
+
+ * [simeon] In MV2, tabs.executeScript defaulted to the active tab; in MV3 scripting.executeScript explicitly requires the target option to be specified.
+ * [oliver] I seem to remember similar changes in other APIs, to make the target of script execution explicit.
+ * [timothy] I see the reasoning there; at the same time it makes the API usage cumbersome for common uses. Perhaps we can achieve the explicitness by supporting a new property instead of tabId?
+ * [simeon] This brings up a broader question around API design philosophy. The change to require a tab ID feels in like with the Extensible Web Manifesto, where lower level and more specific capabilities are provided and the expectation is that the community will produce libraries that provide nicer to use APIs. In our case, though, we don't have a demonstrable history of extension-specific libs being widely adopted. A question I still don't have resolved in my own head is “How batteries-included should extension APIs be?”
+ * [timothy] I'm of the mindset of that we should build the APIs the way they are intended to be used.
+ * [oliver] If we were to add a property such as “useCurrentTab”, do we expect people to understand the risks?
+ * [simeon] We see it in action.onClicked, where people use tabs.query even though the tab is provided in the callback.
+ * [oliver]
```

Comment:
@oliverdunk Do you remember what you said here?

Reply:
Not for definite, but use this 😄 

```suggestion
 * [oliver] I think that may be further motivation to make this explicit, so developers eventually figure out the callback has what they need. As a side note - some APIs already have a [current window](https://developer.chrome.com/docs/extensions/reference/windows/#the-current-window) concept which tries to use the executing context and may be applicable here.
```